### PR TITLE
Game view model refactor

### PR DESCRIPTION
**Pull request by tim-oh-thee-97**

### CODE REVIEW

File: MemoryGame/MemoryGame.csproj [hunk 1]
```diff
@@ -121,38 +122,38 @@
     <None Include="App.config" />
   </ItemGroup>
   <ItemGroup>
-    <Resource Include="Assets\Animals\bear_image.jpg" />
-    <Resource Include="Assets\Animals\deer_image.jpg" />
-    <Resource Include="Assets\Animals\rabbit_image.jpg" />
+    <EmbeddedResource Include="Assets\Animals\bear_image.jpg" />
```
Comment (njrogie) [inline MemoryGame/MemoryGame.csproj:125]: Not sure what the difference is between the embedded resource and the resource.

File: MemoryGame/ResourceManager.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    //method that takes string of name of subfolder (animals, cars food),
+    // checks if folder and assets exist, then returns folder path
+    class ResourceManager
+    {
+        public ResourceManager()
+        {
+            
+        }
+
+        private readonly string OutputDirectory = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData),"MemoryGame");
```
Comment (njrogie) [inline MemoryGame/ResourceManager.cs:25]: Why this directory? I don't see why not, but I'm curious as to why this one was chosen.

File: MemoryGame/Views/GameView.xaml [hunk 3]
```diff
@@ -17,11 +17,11 @@
                 <DockPanel DataContext="{Binding GameInfo}" MinWidth="200" DockPanel.Dock="Right">
                     <StackPanel Visibility="{Binding WinMessage}">
                         <TextBlock Foreground="#FF51FF00">You Won!</TextBlock>
-                        <Button Padding="10" Background="#FF151515" Click="PlayAgain_C" Foreground="#FFA1FFA1" FontSize="20">Play Again</Button>
+                        <Button Padding="10" Background="#FF151515" Command="{Binding PlayAgainCommand}" Foreground="#FFA1FFA1" FontSize="20">Play Again</Button>
                     </StackPanel>
                     <StackPanel Visibility="{Binding LostMessage}">
                         <TextBlock Foreground="#FFFF5050">You Lost!</TextBlock>
-                        <Button Padding="10" Background="#FF151515" Click="PlayAgain_C" Foreground="#FFFF9085" FontSize="20">Play Again</Button>
+                        <Button Padding="10" Background="#FF151515" Command="{Binding PlayAgainCommand}" Foreground="#FFFF9085" FontSize="20">Play Again</Button>
```
Comment (njrogie) [inline MemoryGame/Views/GameView.xaml]: I'm not that familiar with binding vs onclick events. I'd like to discuss this at the next meeting.

File: MemoryGame/Views/GameView.xaml.cs [hunk 4]
```diff
@@ -32,5 +32,11 @@ private void Slide_Clicked(object sender, RoutedEventArgs e)
             var button = sender as Button;
             game.ClickedSlide(button.DataContext as PictureViewModel);
         }
+
+        private void PlayAgain_Click(object sender, RoutedEventArgs e)
```
Comment (njrogie) [inline MemoryGame/Views/GameView.xaml.cs:36]: Is this a re-write of the method from the last commit?

**Review by njrogie — COMMENTED** [review]
Want to discuss my comments at the meeting.

File: MemoryGame/Views/GameView.xaml.cs (hunk 4 above)
Comment (tim-oh-thee-97) [reply MemoryGame/Views/GameView.xaml.cs:36]: Yes, as stated in my most recent commit message.

File: MemoryGame/ResourceManager.cs (hunk 2 above)
Comment (tim-oh-thee-97) [reply MemoryGame/ResourceManager.cs:25]: This is the conventional go-to directory for storing application data. So, when we want to locally store/load the image assets, this is where we would look.

File: MemoryGame/MemoryGame.csproj (hunk 1 above)
Comment (tim-oh-thee-97) [reply MemoryGame/MemoryGame.csproj:125]: For this purpose, they're actually pretty much identical. However, the main difference here is that they are accessed differently, and I only knew how to access them as Embedded Resources from experience.

**Pull request merged**